pkg/render/policyrecommendation.go

```diff
@@ -0,0 +1,329 @@
+	"github.com/tigera/operator/pkg/components"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/secret"
+	"github.com/tigera/operator/pkg/render/common/securitycontext"
+	"github.com/tigera/operator/pkg/tls/certificatemanagement"
+)
+
+// The names of the components related to the PolicyRecommendation APIs related rendered objects.
+const (
+	ElasticsearchPolicyRecommendationUserSecret = "tigera-ee-policy-recommendation-elasticsearch-access"
+
+	PolicyRecommendationName = "tigera-policy-recommendation"
+
+	PolicyRecommendationClusterRoleName        = PolicyRecommendationName
+	PolicyRecommendationClusterRoleBindingName = PolicyRecommendationName
+	PolicyRecommendationDeploymentName         = PolicyRecommendationName
+	PolicyRecommendationNamespace              = PolicyRecommendationName
+	PolicyRecommendationServiceName            = "policy-recommendation"
+	PolicyRecommendationServiceAccountName     = PolicyRecommendationName
+	PolicyRecommendationPolicyName             = networkpolicy.TigeraComponentPolicyPrefix + PolicyRecommendationName
+
+	PolicyRecommendationControllerName = "policy-recommendation-controller"
+	PolicyRecommendationInstallerName  = "policy-recommendation-es-installer"
+
+	PolicyRecommendationServerCertSecret = "tigera-policy-recommendation-server-tls"
+)
+
+// PolicyRecommendationConfiguration contains all the config information needed to render the component.
+type PolicyRecommendationConfiguration struct {
+	ClusterDomain                        string
+	ESClusterConfig                      *relasticsearch.ClusterConfig
+	ESSecrets                            []*corev1.Secret
+	Installation                         *operatorv1.InstallationSpec
+	ManagedCluster                       bool
+	Openshift                            bool
+	PolicyRecommendationServerCertSecret certificatemanagement.KeyPairInterface
+	PullSecrets                          []*corev1.Secret
+	TrustedBundle                        certificatemanagement.TrustedBundle
+
+	// Whether or not the cluster supports pod security policies.
+	UsePSP bool
+}
+
+type policyRecommendationComponent struct {
+	cfg   *PolicyRecommendationConfiguration
+	image string
+}
+
+func PolicyRecommendation(cfg *PolicyRecommendationConfiguration) Component {
+	return &policyRecommendationComponent{
+		cfg: cfg,
+	}
+}
+
+func (pr *policyRecommendationComponent) ResolveImages(is *operatorv1.ImageSet) error {
+	reg := pr.cfg.Installation.Registry
+	path := pr.cfg.Installation.ImagePath
+	prefix := pr.cfg.Installation.ImagePrefix
+
+	var err error
+	pr.image, err = components.GetReference(components.ComponentPolicyRecommendation, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (pr *policyRecommendationComponent) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeAny
+}
+
+func (pr *policyRecommendationComponent) Objects() ([]client.Object, []client.Object) {
+	objs := []client.Object{
```

Comment:
It would be neat to add all objects in this statement that don't require an if-statement.

Reply:
Something like:
```
objs := []client.Object{
	CreateNamespace(PolicyRecommendationNamespace, pr.cfg.Installation.KubernetesProvider, PSSRestricted),
}
objs = append(objs, secret.ToRuntimeObjects(secret.CopyToNamespace(PolicyRecommendationNamespace, pr.cfg.PullSecrets...)...)...)
objs = append(objs,
	pr.serviceAccount(),
	pr.clusterRole(),
	pr.clusterRoleBinding(),
)
objs = append(objs, secret.ToRuntimeObjects(secret.CopyToNamespace(PolicyRecommendationNamespace, pr.cfg.ESSecrets...)...)...)

// Deployment is for standalone or management cluster
if !pr.cfg.ManagedCluster {
	objs = append(objs,
		allowTigeraPolicyForPolicyRecommendation(pr.cfg),
		pr.deployment(),
	)
}
```
or are you thinking of something more condensed?

Reply:
```
	objs := []client.Object{
		CreateNamespace(PolicyRecommendationNamespace, pr.cfg.Installation.KubernetesProvider, PSSRestricted),
		pr.serviceAccount(),
		pr.clusterRole(),
		pr.clusterRoleBinding(),
	}
```